pkg/scheduler/queue/tree_queue_algo_querier_worker_queue_priority.go

```diff
@@ -4,74 +4,54 @@ package queue
+// This distribution of workers across query component subtrees ensures that when one query component is experiencing
+// high latency about 25% of querier-workers continue prioritizing queries for unaffected components.
 //
-// Purpose:
-// This algorithm is intended to ensure that querier-workers are balanced across nodes in the queue selection tree.
-// While it can serve to balance consumers across request queue nodes which have been partitioned by any criteria,
-// its original purpose is to balance querier-workers across queues partitioned a query's expected query component`.
-//
-// There are four possible query components: "ingester", "store-gateway", "ingester-and-store-gateway", and "unknown".
-// With all four possible query component queue nodes active, the modulo operation on the worker IDs
-// will distribute approximately 1 / 4 of the querier-workers to each queue node.
-// A querier-worker ID which is mapped to prioritize the "ingester" queue node will always start there
-// and attempt first to work on a query which only requires the ingesters to complete.
-//
-// By splitting the queues by query component we can ensure that the approximately 1 / 4 of querier-workers are
-// "reserved" for a query component even when the other query component is experiencing high latency.
-// This reservation ensures that the querier-worker connections can continue to process queries
-// which do not need to utilize the query component experiencing high latency.
-//
-// Performance:
-// This significantly outperforms the previous round-robin algorithm which simply rotated through the node order
-// (see TestMultiDimensionalQueueAlgorithmSlowConsumerEffects benchmark outputs for comparison).
+// This significantly outperforms the previous round-robin approach which simply rotated through the node order.
 // Although a vanilla round-robin algorithm will select a given query-component node 1 / 4 of the time,
-// in situations of high latency on a query component that one slow query component will still grow asymptotically
+// in situations of high latency on a query component, the one slow query component will still grow asymptotically
 // to dominate the utilization the querier-worker connections, as measured by inflight query processing time.
 //
-// Implementation Details & Assumptions:
-// The MultiAlgorithmTreeQueue which utilizes this and other QueuingAlgorithm implementations always deletes
-// nodes for paths through the tree which lead to an empty leaf node after a dequeue operation.
-// Nodes for paths through the tree are then re-created when a new request is enqueued which requires that path.
-// This means that a tree will not always have all 4 node types for the 4 possible query component assignments.
+// There are 4 possible query components: "ingester", "store-gateway", "ingester-and-store-gateway", and "unknown".
+// With all 4 queue nodes active, approximately 1 / 4 of the querier-workers are prioritized to each queue node.
```

Comment:
I think the concept of a node being "active" is a little confusing; they either exist or don't, but there's never a component node that exists but is "inactive."
```suggestion
// When all 4 queue nodes exist, approximately 1 / 4 of the querier-workers are prioritized to each queue node.
```

Reply:
done